app/tutorials/example-applications/src/main/java/org/eclipse/mosaic/app/tutorial/emergencybrake/EmergencyBrakeApp.java

```diff
@@ -0,0 +1,220 @@
+import org.eclipse.mosaic.fed.application.app.AbstractApplication;
+import org.eclipse.mosaic.fed.application.app.api.CommunicationApplication;
+import org.eclipse.mosaic.fed.application.app.api.VehicleApplication;
+import org.eclipse.mosaic.fed.application.app.api.os.VehicleOperatingSystem;
+import org.eclipse.mosaic.interactions.communication.V2xMessageTransmission;
+import org.eclipse.mosaic.lib.enums.AdHocChannel;
+import org.eclipse.mosaic.lib.enums.SensorType;
+import org.eclipse.mosaic.lib.geo.GeoPoint;
+import org.eclipse.mosaic.lib.objects.v2x.MessageRouting;
+import org.eclipse.mosaic.lib.objects.v2x.etsi.Denm;
+import org.eclipse.mosaic.lib.objects.v2x.etsi.DenmContent;
+import org.eclipse.mosaic.lib.objects.vehicle.VehicleData;
+import org.eclipse.mosaic.lib.util.objects.ObjectInstantiation;
+import org.eclipse.mosaic.lib.util.scheduling.Event;
+import org.eclipse.mosaic.rti.TIME;
+
+import java.io.File;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * This class implements an application for vehicles.
+ * In case the vehicle sensors detect an obstacle the vehicle will perform an emergency brake.
+ * If the emergency brake endures a specified minimum time duration a DENMessage will be sent out.
+ */
+public class EmergencyBrakeApp extends AbstractApplication<VehicleOperatingSystem> implements VehicleApplication,
+        CommunicationApplication {
+
+    CEmergencyBrakeApp configuration;
+
+    // Keep track of the vehicle movement for the emergency brake detection
+    private float lastSpeed = 0;
+    private float lastTime = 0;
+    private int brakeCounter = 0;
+
+    // Keep status of the emergency brake performed on obstacle
+    private boolean emergencyBrake = false;
+    private long stoppedAt = Long.MIN_VALUE;
+
+    @Override
+    public void onStartup() {
+        initConfig();
+        sample();
+        getOs().getAdHocModule().enable();
+    }
+
+    /**
+     * Creates a new event to sample the emergency brake detection periodically in a specific interval.
+     */
+    private void sample() {
+        long eventTime = getOs().getSimulationTime() + configuration.minimalTimerCallInterval;
+        this.getOs().getEventManager().addEvent(eventTime, this);
+    }
+
+    /**
+     * Gets called periodically to execute the emergency brake detection.
+     *
+     * @param event the event to process
+     * @throws Exception
+     */
+    @Override
+    public void processEvent(Event event) throws Exception {
+        if (!isValidStateAndLog()) {
+            return;
+        }
+        sample();
+        detectEmergencyBrake();
+    }
+
+    /**
+     * Sets all relevant variables from the configuration file.
+     * Note: If the file is corrupted or contains invalid values, default values are assigned.
+     */
+    public void initConfig() {
+        try {
+            configuration = new ObjectInstantiation<>(CEmergencyBrakeApp.class)
```

Comment:
Use `extends ConfigurableApplication` for this class instead of reading the configuration manually here.